internal/util/update/update_test.go

```diff
@@ -789,6 +790,210 @@ func TestCommand_Run_badStrategy(t *testing.T) {
 	}
 }
 
+func TestCommand_Run_subpackages(t *testing.T) {
```

Comment:
I am not sure if we should treat package as a unit here. This might make implementation complicated and might not be the expected behavior. We should deal things more at file level and do 3-way merge. 
for each `filepath` on upstream only, add it to local.
for each `filepath` on upstream and local only, retain local.
for each `filepath` on upstream, origin(initial upstream) and local, do 3-way merge for KRM resources and Kptfile.